signac/contrib/job.py

```diff
@@ -449,46 +449,49 @@ def _init(self, force=False):
         """
         fn_manifest = os.path.join(self._wd, self.FN_MANIFEST)
 
-        # Create the workspace directory if it did not exist yet.
+        # Attempt early exit if the manifest exists and is valid
         try:
-            _mkdir_p(self._wd)
-        except OSError:
-            logger.error(
-                "Error occured while trying to create "
-                "workspace directory for job '{}'.".format(self)
-            )
-            raise
-
-        try:
-            # Ensure to create the binary to write before file creation
-            blob = json.dumps(self._statepoint, indent=2)
+            self._check_manifest()
+        except Exception:
+            # Any exception means this method cannot exit early.
```

Comment:
I am actually wondering whether we should be specific here. Expected exceptions are `OSError`, what else?

Reply:
`_check_manifest` can raise `JobsCorruptedError`, which is a subclass of `RuntimeError`. There may be other exception classes that are raised in that function (it uses `json`, `hashlib`, file reading/decoding). I am hesitant to specify the exception classes because I don't know how to be sure that we're providing sufficient exception coverage and the penalty is high. _Not_ catching an exception would lead to broken behavior, because the job wouldn't actually be initialized at the end of the initialization method.

Reply:
But any of those very unexpected errors are something we should actually raise. We could very easily test which error classes to catch by simply running this against "no file", "corrupted file", "empty file", "binary file", and "permissions wrong", and that should be about it IMO. Catching "all" errors here means to effectively change the behavior of this function, does it not?

Reply:
I would be fine with testing against those cases explicitly. However, I don't think the behavior of the function is changed by catching `Exception` here. From the previous behavior, I would say that the "finishing condition" of the initialization is that `_check_manifest` can complete without raising. I'm just testing that condition at the beginning as a way to exit early. Any unusual errors would still be raised in the original behavior, when `_check_manifest` was called at the end.